src/main/java/com/connexta/poller/service/StatusService.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.service;
+
+import static com.dyngr.core.AttemptResults.continueFor;
+import static com.dyngr.core.AttemptResults.finishWith;
+import static com.dyngr.core.AttemptResults.justContinue;
+
+import com.connexta.store.exceptions.DetailedResponseStatusException;
+import com.dyngr.PollerBuilder;
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.core.StopStrategies;
+import com.dyngr.core.WaitStrategies;
+import java.net.URI;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+/**
+ * This is class is meant to be instantiated as a Bean, Component, or Service. The class's
+ * responsibilities are to manage the settings for polling and kick off new polling tasks. Other
+ * responsibilities include settings the wait and retry behavior. It also creates and configures the
+ * executor, and ensures that it is shutdown properly.
+ */
+@Service
+@Slf4j
+@AllArgsConstructor
+public class StatusService {
+
+  private final int secondsBetweenRetries;
+  private final int secondsToLive;
+  @NotNull ExecutorService executorService;
+  @NotNull WebClient webClient;
+
+  public Future<StatusResponse> poll(URI uri) {
+    final HttpStatus expectedHttpStatus = HttpStatus.OK;
+
+    AttemptMaker<StatusResponse> attemptMaker =
+        () -> {
+          StatusResponse statusResponse = null;
```

Comment:
❗️Add tests for this AttemptMaker

Reply:
How would you recommend testing it? Maybe injecting a mock WebClient and using _verify_ to make sure the right methods were called?

Reply:
First I would extract the body of this lambda to a class or method so it can be tested independently. These are the different tests that I would do:
- mock webclient to throw exceptions
- use a real webclient (WebClient.create()) and mock the server (MockWebServer) to
  - return non-202 error codes
  - return invalid response
  - happy path